saleor/static/dashboard-next/auth/components/SectionRoute.tsx

```diff
@@ -0,0 +1,35 @@
+import * as React from "react";
+import { Route, RouteProps } from "react-router-dom";
+
+import { UserContext } from "..";
+import AppRoot from "../../AppRoot";
+import NotFound from "../../NotFound";
+
+interface SectionRouteProps extends RouteProps {
+  resource?: string;
+}
+
+export const SectionRoute: React.StatelessComponent<SectionRouteProps> = ({
+  resource,
+  ...props
+}) => (
+  <UserContext.Consumer>
+    {({ user }) =>
+      resource ? (
+        user.permissions.map(perm => perm.code).includes(resource) ? (
```

Comment:
It looks like a good candidate for a helper function. If we had one simple function e.g. `checkPermissions(user)` we would use it across the whole project which would greatly improve readability.